### PR TITLE
Improve explanations of what kpt is.

### PR DESCRIPTION
**Pull request by bgrant0607**
As discussed in #3356.

### CODE REVIEW

**Comment by johnbelamaric** [comment]
This is definitely much better. It still reads a bit academic in some places - there's a fair number of assumptions about background knowledge in statements like "This enables interoperability of a variety of generators, transformers, and validators. One doesn’t have to make all changes through a monolithic generator implementation."

Similarly, I think WYSIWYG is great for those of use that have been around a while, but I am not sure most people (especially worldwide) are all that familiar with the term.

None of this blocks the PR, which like I said is much better. But I think we can do a lot more to make this clearer.

**Comment by bgrant0607** [comment]
Thanks @johnbelamaric. I agree with you. I was trying to just update the tagline, add the explanation from my recent twitter thread, and make a few other edits in the time I had. 

I'd be happy to replace WYSIWYG with a term more widely in the current decade, but haven't thought of a good replacement yet. Suggestions?

We also need a glossary.

File: site/faq/README.md [hunk 1]
```diff
@@ -6,13 +6,31 @@ Please visit the [roadmap document] and the [kpt milestones].
...
+
+kpt uses an in-place transformation approach to customization: 
+read configuration in, modify it, and write it back.
+
+This enables interoperability of a variety of generators,
 transformers, and validators. One doesn’t have to make all changes through a monolithic
 generator implementation.
 
+By storing the result of config generators and transformers, automated mutations can be separated in time
+from use and other modifications. This enables generation via a UI, for example.
```
Comment (droot) [inline site/faq/README.md:25]: `separated in time` is difficult to grok :) we might need an example to explain it.

**Review by droot — APPROVED** [review]
This is an area where we will keep on iterating. This is definitely an improvement over what we had, so I am happy merging this.

**Review by yuwenma — APPROVED** [review]

File: site/faq/README.md (hunk 1 above)
Comment (bgrant0607) [reply site/faq/README.md:25]: Ack

**Pull request merged**